src/Numerics/ODESolvers/ODESolvers.jl

```diff
@@ -122,31 +118,15 @@ function solve!(
         time = general_dostep!(
             Q,
             solver,
-            p,
+            param,
             timeend;
             adjustfinalstep = adjustfinalstep,
         )
 
-        # FIXME: Determine better way to handle postcallback behavior
-        # Current behavior:
-        #   retval = 1 exit after all callbacks
-        #   retval = 2 exit immediately
-        retval = 0
-        for (i, cb) in enumerate(callbacks)
-            # FIXME: Consider whether callbacks need anything, or if function closure
-            #        can be used for everything
-            thisretval = cb()
-            thisretval = (thisretval == nothing) ? 0 : thisretval
-            !(thisretval in (0, 1, 2)) &&
-            error("callback #$(i) returned invalid value. It should return either:
-                  `nothing` (continue time stepping)
-                  `0`       (continue time stepping)
-                  `1`       (stop time stepping after all callbacks)
-                  `2`       (stop time stepping immediately)")
-            retval = max(thisretval, retval)
-            retval == 2 && return gettime(solver)
+        val = GenericCallbacks.call!(callbacks, solver, Q, param, t)
+        if val !== nothing && val > 0
```

Comment:
```suggestion
        if !(val === nothing) && val > 0
```

Reply:
any reason why?

Reply:
Isn't the `===` evaluation more efficient than `==` for `nothing`?

Reply:
that would be `!=`, the above was correct for egal

Reply:
yep, `!==(a,b)` is defined as `!(a === b)`.